Update sigmastate (Height is Int)

greenhat opened the pull request:
References:
https://github.com/ScorexFoundation/sigmastate-interpreter/pull/340
https://github.com/ScorexFoundation/sigmastate-interpreter/issues/335

Todo:
- [x] switch `box.creationInfo._1` to Int (height when block got included) in sigmastate and remove `Downcast`;
- [x] find and fix one extra (compared to `voting`) failing test;
- [x] switch to sigmastate's snapshot from `master`;

catena2w commented on src/main/scala/org/ergoplatform/nodeView/state/ErgoState.scala:
```diff
@@ -105,21 +108,21 @@ object ErgoState extends ScorexLogging {
+    val coinsToIssue = If(LT(Height, IntConstant(s.fixedRatePeriod.toInt)),
       s.fixedRate,
-      Minus(s.fixedRate, Multiply(s.oneEpochReduction, epoch))
+      Minus(s.fixedRate.toInt, Multiply(s.oneEpochReduction.toInt, epoch))
```
`fixedRate` and `oneEpochReduction` should remain Long

ergomorphic replied:
@catena2w you mean the operations should be performed with long values? Because header is now Int.

catena2w replied:
height is in, but this does not affect fixedRate and oneEpochReduction, they are still long

greenhat replied:
@ergomorphic @catena2w Done. I've upcasted `epoch` to Long.

greenhat commented:
@kushti I've rebased and updated genesis state hash for it:test. Only two of them are failing, but they're the same two tests that also fail in `voting` branch.
